packages/babel-cli/src/babel/util.js

```diff
@@ -111,3 +112,34 @@ export function adjustRelative(relative, keepFileExtension) {
   }
   return relative.replace(/\.(\w*?)$/, "") + ".js";
 }
+
+export function printSettings() {
+  const hasYarn = cwd =>
+    fs.existsSync(path.resolve(cwd || process.cwd(), "yarn.lock"));
+  const indent = str =>
+    `\n ${str
+      .trim()
+      .split("\n")
+      .join("\n")}`;
+  const log = [];
+  log.push("-------Babel Settings--------");
+  log.push(`Node version: ${indent(process.versions.node)}`);
+  log.push(`Npm version: ${indent(child.execSync("npm -v").toString())}`);
+
+  if (hasYarn()) {
+    log.push(`Yarn version: ${indent(child.execSync("yarn -v").toString())}`);
+  }
+  log.push("-------Babel packages--------");
+  const packages = child
+    .execSync("npm list --silent | grep babel")
```

Comment:
that will catch also not-official babel packages, but actually I think that its fine

Reply:
We talked off but relevant here: I was wondering if npm could be called programmatically? Spawning a new process has some disadvantages like the one you mentioned.

Reply:
+ ouput formating looks a bit strange to me.

Reply:
we can always change the output format.